packages/eas-cli/src/commands/env/delete.ts

```diff
@@ -0,0 +1,127 @@
+import { Flags } from '@oclif/core';
+import chalk from 'chalk';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import {
+  EASEnvironmentFlag,
+  EASNonInteractiveFlag,
+  EASVariableScopeFlag,
+} from '../../commandUtils/flags';
+import { EnvironmentVariableEnvironment, EnvironmentVariableScope } from '../../graphql/generated';
+import { EnvironmentVariableMutation } from '../../graphql/mutations/EnvironmentVariableMutation';
+import { EnvironmentVariablesQuery } from '../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../log';
+import { promptAsync, toggleConfirmAsync } from '../../prompts';
+import { promptVariableEnvironmentAsync } from '../../utils/prompts';
+
+type DeleteFlags = {
+  name?: string;
+  environment?: EnvironmentVariableEnvironment;
+  'non-interactive': boolean;
+  scope?: EnvironmentVariableScope;
+};
+
+export default class EnvironmentVariableDelete extends EasCommand {
+  static override description = 'delete an environment variable by name';
+
+  static override hidden = true;
+
+  static override flags = {
+    name: Flags.string({
+      description: 'Name of the variable to delete',
+    }),
+    ...EASVariableScopeFlag,
+    ...EASEnvironmentFlag,
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    const { flags } = await this.parse(EnvironmentVariableDelete);
+    let { name, environment, 'non-interactive': nonInteractive, scope } = this.validateFlags(flags);
+    const {
+      privateProjectConfig: { projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(EnvironmentVariableDelete, {
+      nonInteractive,
+    });
+
+    if (scope === EnvironmentVariableScope.Project) {
+      if (!environment) {
+        environment = await promptVariableEnvironmentAsync(nonInteractive);
+      }
+    }
+
+    const variables =
+      scope === EnvironmentVariableScope.Project && environment
+        ? (await EnvironmentVariablesQuery.byAppIdAsync(graphqlClient, projectId, environment))
+            .appVariables
```

Comment:
I believe that these can also include `shared` variables, right?

Reply:
One issue I believe the users will run into is a scenario when they do `eas env:delete --scope project` and select a shared variable then we delete it for them in `--non-interactive` mode or prompt them to delete it in standard mode. It can lead to some frustration among users because I believe if you are a new user and you don't understand our concepts and you do `eas env:delete --scope project` you probably want to delete the shared env variable only from this project (unlink it) rather then delete it globally. You probably would like to do so when doing `--scope shared`.

Maybe one should be unable to delete `shared` variables when doing a `--scope project`?